js_modules/dagster-ui/packages/ui-core/src/assets/AssetTable.tsx

```diff
@@ -61,18 +61,26 @@ export const AssetTable = ({
 }: Props) => {
   const [toWipe, setToWipe] = React.useState<AssetKeyInput[] | undefined>();
 
-  const groupedByDisplayKey = groupBy(assets, (a) => JSON.stringify(displayPathForAsset(a)));
-  const displayKeys = Object.keys(groupedByDisplayKey).sort();
+  const groupedByDisplayKey = useMemo(
+    () => groupBy(assets, (a) => JSON.stringify(displayPathForAsset(a))),
+    [assets, displayPathForAsset],
+  );
+  const displayKeys = useMemo(() => Object.keys(groupedByDisplayKey).sort(), [groupedByDisplayKey]);
 
   const [{checkedIds: checkedDisplayKeys}, {onToggleFactory, onToggleAll}] =
     useSelectionReducer(displayKeys);
 
-  const checkedAssets: Asset[] = [];
-  displayKeys.forEach((displayKey) => {
-    if (checkedDisplayKeys.has(displayKey)) {
-      checkedAssets.push(...(groupedByDisplayKey[displayKey] || []));
-    }
-  });
+  const checkedAssets = useMemo(() => {
+    const assets: Asset[] = [];
+    displayKeys.forEach((displayKey) => {
+      if (checkedDisplayKeys.has(displayKey)) {
+        groupedByDisplayKey[displayKey]?.forEach((asset) => {
+          assets.push(asset);
+        });
+      }
+    });
+    return assets;
```

Comment:
How about this?

```
return displayKeys
  .filter((displayKey) => checkedDisplayKeys.has(displayKey))
  .map((displayKey) => groupedByDisplayKey[displayKey] || [])
  .flat();
```

Reply:
Eh, I rather skip the intermediate array 😬